### PR TITLE
312 - refactor git repo object

### PR DESCRIPTION
**Pull request by mbao01**
_Merge Order - 1_

### CODE REVIEW

**Comment by mbao01** [comment]
@yegor256 please review

**Comment by mbao01** [comment]
@yegor256 please review

File: 0pdd.rb [hunk 1]
```diff
@@ -213,11 +213,24 @@
...
+      id_rsa: settings.config['id_rsa'],
+      dir: settings.temp_dir,
+      master: master || 'master'
+    )
+    repo.push
+    xml = repo.xml
+    xml.xpath('//processing-instruction("xml-stylesheet")').remove
+    xml.to_s
+  rescue StandardError
+    error 400, "Could not get snapshot for #{name}"
```
Comment (yegor256) [inline 0pdd.rb:232]: @mbao01 I still don't understand why we need this. You catch 'StandardError' which can mean anything and then make a decision that it meant that the repo is not found. Maybe the database connection is lost? Maybe out of memory? Why "not found"?

**Comment by yegor256** [comment]
@mbao01 see above

File: 0pdd.rb (hunk 1 above)
Comment (mbao01) [reply 0pdd.rb:232]: @yegor256 No, you are right. The error I meant to catch here is `Exec:Error`. I updated this.

**Comment by mbao01** [comment]
@yegor256 please review

File: objects/git_repo.rb [hunk 2]
```diff
@@ -100,9 +100,9 @@ def pull
         "cd #{@path}",
         "master=#{Shellwords.escape(@master)}",
         'git config --local core.autocrlf false',
-        'git fetch --force --all --quiet',
-        'git reset --hard --quiet origin/${master}',
+        'git reset origin/${master} --hard --quiet',
         'git clean --force -d',
+        'git fetch --quiet',
```
Comment (yegor256) [inline objects/git_repo.rb:105]: @mbao01 why did you remove `--force --all`? I believe, they are here for a reason.

File: 0pdd.rb [hunk 3]
```diff
@@ -213,11 +213,24 @@
...
+      name: name,
+      id_rsa: settings.config['id_rsa'],
+      dir: settings.temp_dir,
+      master: master || 'master'
+    )
+    repo.push
+    xml = repo.xml
+    xml.xpath('//processing-instruction("xml-stylesheet")').remove
+    xml.to_s
+  rescue Exec::Error => e
```
Comment (yegor256) [inline 0pdd.rb:231]: @mbao01 can you please explain WHY do you need special error handling in this particular web entry point? why not in all others?

**Comment by yegor256** [comment]
@mbao01 see above

File: objects/git_repo.rb (hunk 2 above)
Comment (mbao01) [reply objects/git_repo.rb:105]: @yegor256 It seems like a simple change. I would not touch this if I did not encounter an issue with it while working on this integration. Maybe a good thing will be to leave it as is and just add a @todo marker. Please let me know how to proceed.

**Comment by mbao01** [comment]
@yegor256 please try again

**Comment by yegor256** [comment]
@rultor merge
